src/actions/sources/newSources.js

```diff
@@ -46,20 +47,24 @@ function createOriginalSource(
   };
 }
 
-function loadSourceMaps(sources) {
-  return async function({ dispatch, sourceMaps }: ThunkArgs) {
+function loadSourceMaps(sources: Source[]) {
+  return async function({
+    dispatch,
+    sourceMaps
+  }: ThunkArgs): Promise<Promise<Source>[]> {
     if (!prefs.clientSourceMapsEnabled) {
-      return;
+      return [];
     }
 
-    let originalSources = await Promise.all(
-      sources.map(({ id }) => dispatch(loadSourceMap(id)))
+    return flatten(
```

Comment:
we don't really use the return value of this function, but it is nice to keep it as an array of original sources for debugging purposes.